services/time/src/lib.rs

```diff
@@ -111,62 +113,100 @@ transactions! {
     }
 }
 
-impl Transaction for TxTime {
-    fn verify(&self) -> bool {
-        self.verify_signature(self.pub_key())
-    }
+#[derive(Debug, Fail)]
+#[repr(u8)]
+enum Error {
+    #[fail(display = "Not authored by a validator")]
+    UnknownSender = 0,
 
-    fn execute(&self, view: &mut Fork) -> ExecutionResult {
-        let validator_keys = Schema::new(&view).actual_configuration().validator_keys;
+    #[fail(display = "The validator time is longer than the proposed one")]
```

Comment:
I don't think "longer" can be used in this context. Perhaps, "greater" would be better? @Maria-Nosyk